c10/core/DispatchKeySet.h

```diff
@@ -74,31 +252,63 @@ class DispatchKeySet final {
     return DispatchKeySet(repr_ | other.repr_);
   }
   // Perform set intersection
-  DispatchKeySet operator&(DispatchKeySet other) const {
+  constexpr DispatchKeySet operator&(DispatchKeySet other) const {
     return DispatchKeySet(repr_ & other.repr_);
   }
-  // Compute the set difference self - other
+  // Compute the set difference self - other,
+  // but ONLY for the functionality keys.
+  // Any backend bits set on self will remain unchanged.
+  // See Note [Removing keys from DispatchKeySet Only Affects Functionality Keys]
   DispatchKeySet operator-(DispatchKeySet other) const {
-    return DispatchKeySet(repr_ & ~other.repr_);
+    return DispatchKeySet(repr_ & (full_backend_mask | ~other.repr_));
+  }
+  // Perform a right bitshift
+  DispatchKeySet operator>>(uint8_t shift) const {
```

Comment:
You're right, I'll delete it. I was looking for where I ended using it, and the place I used it really isn't necessary:
```
auto backend_idx = ((ks & DispatchKeySet(DispatchKeySet::RAW, full_backend_mask)) >> 1).indexOfHighestBit();
```
I had to offset things by 1 because we don't want to include an extra slot for `BackendComponent::InvalidBit`. But I can do the offset just as easily with:
```
auto backend_idx = (ks & DispatchKeySet(DispatchKeySet::RAW, full_backend_mask)).indexOfHighestBit() - 1;
```